### DISCUSSION TITLE
Ratings component issue with half masks

### DISCUSSION THREAD

**Discussion by boian-ivanov** [saadeghi/daisyui#1567] [Q&A] [answered]
Not sure if that's is an implementation issue on my side or if it's to do with the masks themselves, but when creating a star rating with half stars (it works flawlessly with full ones), I get a lack of in-between space on mobile devices and sometimes it's also replicable on desktop as well. I'm adding reproductions bellow of how it looks and the code that generates them as well.
<img width="271" alt="Desktop reproduction in responsive variant" src="https://user-images.githubusercontent.com/15968453/215512002-48aee36f-6cb5-4860-9ec6-4d8081b1110a.png">
This occurs from what I've seen from width of 768 to 784 in my example, but then it's always like this on a mobile device (both Chrome & Safari on iOS)
![Mobile representation](https://user-images.githubusercontent.com/15968453/215512640-478353e3-b752-464e-a72a-59d161d76125.jpeg)

So I've been wondering what might be causing this issue. The markup for this exact bug is the following (with checked being applied for the first 7 in the above example):
```html
<div class="rating rating-half">
  <input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-1"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-2"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-1"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-2"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-1"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-2"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-1"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-2"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-1"
    disabled=""
  /><input
    type="radio"
    name="rating-0.5804767042521979"
    class="mask mask-star bg-accent cursor-default mask-half-2"
    disabled=""
  />
</div>
```

Also, this is even worse when using the second available mask. Any ideas?

**Comment by saadeghi** [comment, marked as answer]
Thank you for reporting. iOS adds a border-radius by default.  
It is now fixed in the last version of daisyUI: `2.50.0`

**Reply by boian-ivanov** [reply]
That has fixed it, so thank you so much for your quick response! 👏 
But there's a strange issue occurring on all browsers with half-stars, where you would get a line in between each star (or between each half-rating) and I cannot seem to understand what's causing it. In my mind, it might be to do with using the half mask for left and right and then sticking them together. I wonder if the behaviour might improve if there were 3 svg types, one for empty star, half-full and full. This way there wouldn't be a need to stick them together and this might not occur anymore, but that's mostly thinking out loud. 
Here's the issue I'm talking about:

**Mobile (Safari on iOS 16.x)** 

<img width="245" alt="image" src="https://user-images.githubusercontent.com/15968453/216821826-a9da9cc4-e15d-4d1f-91d1-9a770f4e4866.jpeg">

**Desktop (Chrome 108.x on MacOS)** 

<img width="245" alt="image" src="https://user-images.githubusercontent.com/15968453/216821867-6b8e6709-4fe9-4500-85d3-886074d066fb.png">

Both are rendering the `mask-star-2` variation, but same happens with `mask-star` (as seen on the initial post), so I assume it's to do with the way browsers show the split version of the mask image svg, but I could be wrong.

**Reply by saadeghi** [reply]
> <img alt="image" width="245" src="https://user-images.githubusercontent.com/15968453/216821867-6b8e6709-4fe9-4500-85d3-886074d066fb.png">

That's strange. Do you see the same issue [here](https://play.tailwindcss.com/i8oJGXANSI) as well?

This shouldn't happen because it's just 2 SVGs next to each other without any space between. And I can't see this issue on my Chrome (MacOS) or Safari (iOS) 
Your screenshot looks like it has a space but it's less than a pixel so you couldn't find the root of issue, a hotfix would be adding a class like `-space-x-[0.1px]` to `rating`

**Reply by boian-ivanov** [reply]
I see what you mean. I've tried the playground and I can't for the life of me make it appear. For me this is a browser issue, but then again, if it's not appearing in the playground, it then seems like it could be an issue on my side. I'll play around with it, but I see what you mean with the spacing difference, although in order to completely remove it through it, I need more like `0.3px` and then it goes in the other direction, so not ideal.
Thank you for the suggestion, it's definitely not an issue with Daisy, but then I'm curious may be causing it 🤔

**Reply by jamestolley** [reply]
I'm seeing this as well, with 4.5.0. Any idea what caused it for you? Thanks

**Reply by jamestolley** [reply]
I think it's due to a css transform / scale of the container.
